SteamBot/Logging/Log.cs

```diff
@@ -95,10 +99,22 @@ public void RemoveLoggingObject(LoggerBase logger)
 
         public void Dispose()
         {
-            Disposed = true;
-            foreach (IDisposable logger in LoggerObjects.OfType<IDisposable>())
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!Disposed)
             {
-                logger.Dispose();
+                if (disposing)
+                {
+                    foreach (IDisposable disposableLogger in LoggerObjects.OfType<IDisposable>())
+                        disposableLogger.Dispose();
```

Comment:
I don't understand the question.  You're calling `Dispose()` on the `IDisposable` resources, which is correct, and which means you are not making that assumption.